qtensor/tools/lightcone_orbits.py

```diff
@@ -45,28 +48,36 @@ def graph_cert(G):
     cert = pynauty.certificate(g)
     return cert
 
+def get_cert_e_tuples(e,G=None,p=None):
+    """A helper function for multiprocessing
+    """
+    subgraph = relabel_edge_first(get_edge_subgraph(G, e, p), e)
+    cert = graph_cert(subgraph)
+    return e,cert
+
 
 def get_edge_orbits_lightcones(G,p):
     """Takes graph G and number of QAOA steps p
     returns unique subgraphs that QAOA sees
     dict: {orbit_id : [list of edges in orbit]} 
     and maximum number of nodes in a lightcone subgraph
-    if maxnnodes == G.number_of_nodes(), this simply becomes edge orbits
     """
-    maxnnodes = -1
 
     eorbits = defaultdict(list)
     # for each edge construct the light cone subgraph and compute certificate  
-    for e in G.edges():
-        subgraph = relabel_edge_first(get_edge_subgraph(G, e, p), e)
-        cert = graph_cert(subgraph)
+    if G.number_of_edges() > 1000:
+        # accelerate with multiprocessing if computing for a large graph
+        with Pool(psutil.cpu_count()) as pool:
+            certs_e_tuples = pool.map(partial(get_cert_e_tuples, G=G,p=p), G.edges())
+    else:
+        certs_e_tuples = [get_cert_e_tuples(e,G=G,p=p) for e in G.edges()]
+
+    for e,cert in certs_e_tuples:
         eorbits[cert].append(e)
 
     eorbits_integer_keys = {}
     for i, (cert, edges) in enumerate(eorbits.items()):
         eorbits_integer_keys[i] = edges
 
     assert(len(eorbits_integer_keys) == len(eorbits))
-    return eorbits_integer_keys, maxnnodes
-
-
+    return eorbits_integer_keys, None 
```

Comment:
What's the point of this second return value then? Is it used anywhere?

Reply:
It is not. Removed.